packages/aws-cdk-lib/aws-codepipeline/lib/trigger.ts

```diff
@@ -116,11 +166,32 @@ export class Trigger {
       }
 
       pushFilter?.forEach(filter => {
+        if (!filter.branchesExcludes && !filter.branchesIncludes && (filter.filePathsExcludes || filter.filePathsIncludes)) {
+          throw new Error(`cannot specify filePaths without branches in pushFilter for sourceAction with name '${sourceAction.actionProperties.actionName}'`);
+        }
```

Comment:
If there is no this validation, the error occurred in CFn: `The trigger configuration for the source action is not valid. Make sure to choose valid filter combinations.`.

In addition, the file paths in the `Branch` is optional in the AWS console.